test/unit/util/locking/RedisResourceLocker.test.ts

```diff
@@ -0,0 +1,169 @@
+  beforeEach(async(): Promise<void> => {
+    // eslint-disable-next-line import/no-named-as-default-member
+    redis.createClient = jest.fn().mockReturnValue({
+      on: jest.fn(),
+      quit: jest.fn(),
+      get: jest.fn(),
+    });
+
+    locker = new RedisResourceLocker([ '6379' ]);
+    locker.redlock.lock = jest.fn().mockImplementation(
+      async(resource: string, ttl: number): Promise<Lock> =>
+        new Lock(locker.redlock, resource, 'test', ttl, 100),
+    );
+    locker.redlock.unlock = jest.fn();
+    locker.redlock.extend = jest.fn().mockImplementation(
+      async(lock: Lock, ttl: number): Promise<Lock> => {
+        lock.expiration += ttl;
+        return lock;
+      },
+    );
+  });
+
+  it('can lock and unlock a resource.', async(): Promise<void> => {
+    await expect(locker.acquire(identifier)).resolves.toBeUndefined();
+    await expect(locker.release(identifier)).resolves.toBeUndefined();
+    expect(locker.redlock.lock).toHaveBeenCalledTimes(1);
+    expect(locker.redlock.unlock).toHaveBeenCalledTimes(1);
+  });
+
+  it('can lock a resource again after it was unlocked.', async(): Promise<void> => {
+    await expect(locker.acquire(identifier)).resolves.toBeUndefined();
+    await expect(locker.release(identifier)).resolves.toBeUndefined();
+    await expect(locker.acquire(identifier)).resolves.toBeUndefined();
+    expect(locker.redlock.lock).toHaveBeenCalledTimes(2);
+    expect(locker.redlock.unlock).toHaveBeenCalledTimes(1);
+    await expect(locker.release(identifier)).resolves.toBeUndefined();
+  });
+
+  it('errors when unlocking a resource that was not locked.', async(): Promise<void> => {
+    await expect(locker.acquire(identifier)).resolves.toBeUndefined();
+    await expect(locker.release(identifier)).resolves.toBeUndefined();
+    await expect(locker.release(identifier)).rejects.toThrow(InternalServerError);
+    expect(locker.redlock.lock).toHaveBeenCalledTimes(1);
+    expect(locker.redlock.unlock).toHaveBeenCalledTimes(1);
+  });
+
+  it('errors when redlock.lock throws an error.', async(): Promise<void> => {
+    locker.redlock.lock = jest.fn().mockRejectedValue(new Error('random Error'));
+    await expect(locker.acquire(identifier)).rejects.toThrow(InternalServerError);
+    await expect(locker.acquire(identifier)).rejects.toThrow('Unable to acquire lock for ');
+    await expect(locker.acquire(identifier)).rejects.toThrow('Error: random Error');
+    expect(locker.redlock.lock).toHaveBeenCalledTimes(3);
+  });
+
+  it('errors when redlock.unlock throws an error.', async(): Promise<void> => {
+    await locker.acquire(identifier);
+    locker.redlock.unlock = jest.fn().mockRejectedValue(new Error('random Error'));
+    await expect(locker.release(identifier)).rejects.toThrow(InternalServerError);
+    await expect(locker.release(identifier)).rejects.toThrow('Unable to release lock for: ');
+    await expect(locker.release(identifier)).rejects.toThrow('Error: random Error');
```

Comment:
```suggestion
    const prom = locker.release(identifier);
    await expect(prom).rejects.toThrow(InternalServerError);
    await expect(prom).rejects.toThrow('Unable to release lock for: ');
    await expect(prom).rejects.toThrow('Error: random Error');
```
This way you don't need to run the function 3 times. Should also be possible to combine the last 2 tests in a single string right?